packages/flutter/lib/src/cupertino/text_selection_toolbar.dart

```diff
@@ -278,6 +278,26 @@ class _RenderCupertinoTextSelectionToolbarShape extends RenderShiftedBox {
 
   bool get isAbove => anchorAbove.dy >= (child?.size.height ?? 0.0) - _kToolbarArrowSize.height * 2;
 
+  Offset _computeChildOffset(Size childSize) {
+    final bool isAbove = anchorAbove.dy >= childSize.height - _kToolbarArrowSize.height * 2;
+    return Offset(0.0, isAbove ? -_kToolbarArrowSize.height : 0.0);
+  }
+
+  @override
+  double? computeDryBaseline(covariant BoxConstraints constraints, TextBaseline baseline) {
+    final RenderBox? child = this.child;
+    if (child == null) {
+      return null;
+    }
+    final BoxConstraints enforcedConstraint = BoxConstraints(
+      minWidth: _kToolbarArrowSize.width + _kToolbarBorderRadius.x * 2,
+    ).enforce(constraints.loosen());
```

Comment:
Also wondering if we should deduplicate this with performLayout and have a method that gives us the child constraints based on provided constraints?